src/Engine/Combat/BattleInstance.java

```diff
@@ -45,10 +48,10 @@ public BattleInstance(Unit pAttacker, Unit pDefender, GameMap map, int attackerX
     if( 1 == battleRange )
     {
       defenderWeapon = defender.chooseWeapon(attacker.model, battleRange, false);
-      if( null != defenderWeapon )
-      {
-        canCounter = true;
-      }
+    }
+    else
+    {
+      defenderWeapon = null;
```

Comment:
Because I made `defenderWeapon` final, it wouldn't let me change it if it was initialized to null. By the same token, it *required* that all code paths set the value.
I could just make it not final, but I want it final.